test/unit-test/vp8/vp8_utest.c

```diff
@@ -432,15 +592,153 @@ void test_VP8_Packetizer_AllProperties( void )
+                                 &( frame ) );
+
+    TEST_ASSERT_EQUAL( VP8_RESULT_BAD_PARAM,
+                       result );
+
+    frame.pFrameData = &( frameData[ 0 ] );
+    frame.frameDataLength = 0;
+
+    result = VP8Packetizer_Init( &( ctx ),
+                                 &( frame ) );
 
+    TEST_ASSERT_EQUAL( VP8_RESULT_BAD_PARAM,
+                       result );
+}
 
+/*-----------------------------------------------------------*/
 
 /**
- * @brief Validate VP8 depacketization happy path.
+ * @brief Validate VP8_Packetizer_GetPacket in case of bad parameters.
+ */
+void test_VP8_Packetizer_GetPacket_BadParams( void )
+{
+    VP8Result_t result;
+    uint8_t frameData[] = { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05,
+                            0x10, 0x11, 0x12, 0x13, 0x14, 0x15,
+                            0x20, 0x21, 0x22, 0x23, 0x24, 0x25,
+                            0x30, 0x31, 0x32, 0x33, 0x34, 0x35,
+                            0x40, 0x41 };
+    VP8PacketizerContext_t ctx = { 0 };
+    VP8Frame_t frame;
+    VP8Packet_t pkt;
+
+    memset( &( frame ),
+            0,
+            sizeof( VP8Frame_t ) );
+
+    frame.pFrameData = &( frameData[ 0 ] );
+    frame.frameDataLength = sizeof( frameData );
+
+    result = VP8Packetizer_Init( &( ctx ),
+                                 &( frame ) );
+
+    TEST_ASSERT_EQUAL( VP8_RESULT_OK,
+                       result );
+
+    pkt.pPacketData = &( packetBuffer[ 0 ] );
+    pkt.packetDataLength = PACKET_BUFFER_LENGTH;
+
+    result = VP8Packetizer_GetPacket( NULL,
+                                      &( pkt ) );
+
+    TEST_ASSERT_EQUAL( VP8_RESULT_BAD_PARAM,
+                       result );
+
+    result = VP8Packetizer_GetPacket( &( ctx ),
+                                      NULL );
+
+    TEST_ASSERT_EQUAL( VP8_RESULT_BAD_PARAM,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate VP8_Packetizer_GetPacket incase of out of memory.
  */
+void test_VP8_Packetizer_GetPacket_OutOfMemory( void )
+{
+    VP8Result_t result;
+    uint8_t frameData[] = { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05,
+                            0x10, 0x11, 0x12, 0x13, 0x14, 0x15,
+                            0x20, 0x21, 0x22, 0x23, 0x24, 0x25,
+                            0x30, 0x31, 0x32, 0x33, 0x34, 0x35,
+                            0x40, 0x41 };
+    VP8PacketizerContext_t ctx = { 0 };
+    VP8Frame_t frame;
+    VP8Packet_t pkt;
+
+    memset( &( frame ),
+            0,
+            sizeof( VP8Frame_t ) );
+
+    frame.pFrameData = &( frameData[ 0 ] );
+    frame.frameDataLength = sizeof( frameData );
+
+    result = VP8Packetizer_Init( &( ctx ),
+                                 &( frame ) );
+
+    TEST_ASSERT_EQUAL( VP8_RESULT_OK,
+                       result );
+
+    pkt.pPacketData = &( packetBuffer[ 0 ] );
+    pkt.packetDataLength = PACKET_BUFFER_LENGTH;
+    ctx.payloadDescLength = PACKET_BUFFER_LENGTH + 1;
```

Comment:
I think this is setting the needed memory size fo a packet, is it?

Reply:
Yes, so the actual payload length is 1 more than what can be packetized which gives us "out of memory" output.